### ISSUE TITLE
direct comparison with a little error

### ISSUE THREAD

**Issue by 3Dietrich** [dinther/pathbuilder#11]
I tried further, sometimes OpenSCAD froze, sometimes it printed without end errors until I had to quit it immediately and sometimes it worked... very mixed.

**My approach is like this:** 
- one (or more) object in Inkscape, 
- object selected, go to XML editor 
- click the line "d", copy the content and 
- paste it in OpenSCAD as a variable. 

It has the great advantage that you can work with many objects in Inkscape but use selected objects for OpenSCAD (in contrast to using .svg: unfortunately nothing can be filtered out, i need to delete every other objects). But it has the "disadvantage" that I have to take the object as I see it, whether it is rx*2 or rx*3 ;) it looks correct, but pathfinder gets problems with it. It often just froze. With the svg import I do not have such problems.

**long story short:**
I have found one special difference what _may_ be connected with the freezes/neverending errors:
I built a single shape for testing as svg and pathbuilder. Pathbuilder can render that - with an interesting detail I think. Have a look at the scad file:
[testform_1.zip](https://github.com/dinther/pathbuilder/files/7274321/testform_1.zip)
is this maybe the rx*2>d problem?

**Comment by dinther** [comment]
Yes, I can see the error. I will sort it out.

**Comment by dinther** [comment]
I sat down three times to sort it out but I am having trouble concentrating the last few days.  I see the error so it should be easy enough.

![image](https://user-images.githubusercontent.com/1192916/136117644-a3d0d00f-8477-482d-8ea9-5adfcd047177.png)

I did think that your idea to make a towel clip using path commands is a really good example. So I hand coded a version (Not perfect, I want to do it again) but I thought you might like to see how efficient path code can be.

![image](https://user-images.githubusercontent.com/1192916/136120795-0796cfbe-aaa9-43cd-8241-2bee1bdd2134.png)


```
include <../pathbuilder.scad>
$fa=3; $fs=0.3; $pb_spline=36;

thickness = 7;


union(){
    //clip
    linear_extrude(thickness) svgShape($pb_spline=144, "M55-32.5C-22.5-25 -22.5 25 55 32.5a2.5 2.5 0 0 0 0 -5C-15 20 -15 -20 55 -27.5a2.5 2.5 0 0 0 0-5");
    //arm path 
    arm = "M11 16 10 13.05 12 14.55 fillet1L13 13 C27.5 -8 22.5 -24.5 87.5 -11 l 0 5 C33.5 -11 40 -10 23.9 20.7 fillet1L 25.4 21.2 23.5 21.5z";
    translate([0,0,thickness-thickness*0.48]) linear_extrude(thickness*0.48) svgShape(arm);
    mirror ([0,1])linear_extrude(thickness*0.48) svgShape(arm);
    //teeth1
    linear_extrude(thickness) svgShape("M67.25 -14.75q7.5 0.95 20.25 3.5forward 10 fillet3.5 l-21.1 12.1 0.85 -8.45 -12.25 7.75");
    //teeth2
    linear_extrude(thickness) svgShape($pb_spline=18, "M55 15.75q14.4 -0.75 33 -4.75forward 10 fillet5.5 l-11.6 -11.05 -10.85 6.95 1.2 -9 -13.2 8.2 1.3 -7.85q-6.5 -0.5 -10.15 11.95");
}
```

Alternatively, because I don't want to hold you up, you could create a simple svg text file with the following content:
```
<svg xmlns="http://www.w3.org/2000/svg">
    <path d="m0 0h8v8z"/>
</svg>
```
You can omit the viewBox and size attributes because openSCAD only performs scaling with it.
Replace the path string with yours and import the svg using the openSCAD built in SVG logic.
You can do this with a separate file for each discrete part.


like this.
```
<svg xmlns="http://www.w3.org/2000/svg">
  <path d="M 11.451031,5.538032e-4 C 10.947718,-0.0097662 10.225622,0.1242058 8.7359888,0.3941388 c -2.979268,0.539863 -4.0223631,-0.132797 -5.9082061,0.454307 -1.040364,0.323881 -2.72806402,0.558266 -2.82559502,1.643504 -0.03717,0.413624 0.403802,0.354729 0.98070799,1.197023 0.57690603,0.842293 0.21935103,2.324493 1.22862903,2.324493 0.990976,0 0.651369,-2.04346 1.510176,-2.537219 2.1885691,-1.258276 5.4915121,-1.423659 7.5731813,0.0045 2.133536,1.463782 0.307582,6.5399482 2.798288,7.2406802 0.71267,0.200502 -0.164902,-1.3525722 1.763288,-1.3502342 1.928192,0.0023 1.034178,0.9339372 1.663039,1.2045272 1.449242,0.623586 3.081769,0.719781 4.651887,0.87432 2.204572,0.216987 4.437911,0.259257 6.644996,0.06955 3.070996,-0.263955 6.300211,-0.372097 9.094501,-1.6731092 0.915029,-0.426035 2.122946,-0.940173 2.336049,-1.926768 0.200734,-0.929368 -0.185015,-1.977235 -0.817748,-2.798756 l -5.255998,2.971227 0.641188,-5.019074 -6.306997,5.017889 0.664679,-5.211799 -4.856908,3.274535 c -0.393719,0.674971 -0.836926,1.279869 -1.519105,1.553103 -1.226365,0.491199 -2.72885,0.129414 -3.944511,-0.387703 -0.784182,-0.333577 -1.354972,-1.04439 -1.944639,-1.659624 -0.763142,-0.796223 -1.359941,-1.736964 -2.039098,-2.605932 -0.672294,-0.86019 -1.038772,-2.093559 -2.01535,-2.581616 C 12.238584,0.1651818 11.954342,0.0108378 11.45103,5.258032e-4 Z"/>
</svg>
```

In inkscape you can also save as "Plain svg" it's still not as compact as above but it might save you some hassle.
Although I want my code to be close to compliant with SVG standards. It wasn't really the intent of the tool. I just wanted an easier way to hand code complex parametric shapes. I will fix that bug though.

**Comment by 3Dietrich** [comment]
Thank you for investigating! The base idea comes from [my old Clamp](https://www.prusaprinters.org/prints/40665-yet-another-towel-clip) and [the better one](https://www.thingiverse.com/thing:4934574)  I found.
But now I want to integrate the metal spring of the wooden clamp.
![IMG_3150](https://user-images.githubusercontent.com/4608796/136285006-bd79ad2d-c3f7-43cf-96e9-90e2a0019f5d.jpeg)

Thank you for the tip with the "html - svg file" trick! That's an excellent tip for "isolating" shapes (using the old svg import in OpenSCAD). I tried also "Save as Plain SVG" but this is the same problem as before: it saves all content and OpenSCAD. However, I still need your pathbuilder [BOSL2](https://github.com/revarbat/BOSL2/) !
I thought I found the problem: one node where the line runs crazy are actually 2 nodes lying on top of each other. 
<img width="311" alt="Bildschirmfoto 2021-10-06 um 10 27 44" src="https://user-images.githubusercontent.com/4608796/136276888-17f0c640-bfc1-4536-ab6b-5eca207bdff4.png">
But 'correcting' this doesn't help. Also changing this and neighbor nodes to "auto smooth" or "symetric" nodes doesn't help.. (my guess was that maybe inkscape makes error compensation and this changes cause a reset/correction of the path)
So.. the path(s) seem correct, but still the 'strange outlier':[svgtest_corrections.zip](https://github.com/dinther/pathbuilder/files/7297147/svgtest_corrections.zip)

**Comment by dinther** [comment]
Problem fixed.
